src/ui/main_window/ui_events.cpp

```diff
@@ -175,6 +175,73 @@ void MainWindow::link_views_toggle()
     link_views_enabled_ = !link_views_enabled_;
 }
 
+void MainWindow::shift_precision_left()
+{
+    const auto shift_precision_left = [](Stage* stage) {
```

Comment:
better avoid name shadowing
```suggestion
    const auto shift_precision_left_impl = [](Stage* stage) {
```